maestro-cli/src/main/java/maestro/cli/runner/MaestroCommandRunner.kt

```diff
@@ -31,37 +33,69 @@ import maestro.orchestra.MaestroCommand
 import maestro.orchestra.Orchestra
 import maestro.orchestra.OrchestraAppState
 import maestro.orchestra.yaml.YamlCommandReader
+import org.slf4j.LoggerFactory
+import java.io.File
 import java.util.IdentityHashMap
 
 object MaestroCommandRunner {
 
-    private val logger = DebugLogStore.loggerFor(MaestroCommand::class.java)
+    private val logger = LoggerFactory.getLogger(MaestroCommandRunner::class.java)
 
     fun runCommands(
         maestro: Maestro,
         device: Device?,
         view: ResultView,
         commands: List<MaestroCommand>,
         cachedAppState: OrchestraAppState?,
+        debug: FlowDebugMetadata
     ): Result {
         val initFlow = YamlCommandReader.getConfig(commands)?.initFlow
 
         val commandStatuses = IdentityHashMap<MaestroCommand, CommandStatus>()
         val commandMetadata = IdentityHashMap<MaestroCommand, Orchestra.CommandMetadata>()
 
+        // debug
+        val debugCommands = debug.commands
+        val debugScreenshots = debug.screenshots
+
+        fun takeDebugScreenshot(status: CommandStatus): File? {
+            val containsFailed = debugScreenshots.any { it.status == CommandStatus.FAILED }
+
+            // Avoids duplicate failed images from parent commands
+            if (containsFailed && status == CommandStatus.FAILED) {
+                return null
+            }
```

Comment:
Will this coincide with what Felipe worked on recently surfacing screenshots?

Reply:
This will work only on CLI. Maestro Cloud uses Orchestra directly to run commands which bypasses MaestroCommandRunner.